src/core/connection/SSDConverter.hpp

```diff
@@ -0,0 +1,211 @@
+#pragma once
+
+#include "base/Qv2rayBase.hpp"
+#include "common/QvHelpers.hpp"
+
+namespace Qv2ray::core::handlers
+{
+    /**
+     * A Naive SSD Decoder for Qv2ray
+     *
+     * @author DuckSoft <realducksoft@gmail.com>
+     * @copyright Licensed under GPLv3.
+     */
+    namespace ssd
+    {
+        /**
+         * pattern for the nodes in ssd links.
+         *
+         * %1: airport name
+         * %2: node name
+         * %3: rate
+         */
+        inline auto DEFAULT_NAME_PATTERN = "%1 - %2 (rate %3)";
+
+        /**
+         * @brief decodeSSD
+         * @param uri     the uri of ssd link.
+         * @param pattern the pattern for node names.
+         * @return tuple of:
+         *          - tuple of:
+         *             - airport name (for the sake of grouping)
+         *             - shadowsocks configuration list
+         *          - log list
+         *         in case of error, no objects will be returned.
+         */
+        std::pair <std::optional<std::pair<QString, QList<std::pair<QString, ShadowSocksServerObject>>>>, QStringList>
+        decodeSSD(const QString &uri, const QString &pattern = DEFAULT_NAME_PATTERN);
```

Comment:
I think the return type is quite complicated.
Just return **a tuple of `QMap<QString (connection name), CONFIGROOT>` and a `QString (airport name)`** while using the pointer parameter to store log messages should be ok.

Reply:
With QMap, the items are always sorted by key. I want to avoid this.
Pointer parameter is a good idea. I will give a try.

Reply:
You may want QHash? (I just meant to use a mapping instead of a list of tuples)